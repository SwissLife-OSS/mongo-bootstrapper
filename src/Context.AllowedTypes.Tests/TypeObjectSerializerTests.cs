using MongoDB.Extensions.Context.Internal;
using Snapshooter.Xunit;
using Xunit;
using Xunit.Priority;

namespace MongoDB.Extensions.Context.AllowedTypes.Tests;

[TestCaseOrderer(PriorityOrderer.Name, PriorityOrderer.Assembly)]
public class TypeObjectSerializerTests
{
    [Fact]
    public void AddAllowedTypes_AddAllowedTypesOfAllDependencies_Success()
    {
        // Arrange
        TypeObjectSerializer.Clear();

        // Act
        TypeObjectSerializer.AddAllowedTypesOfAllDependencies();

        // Assert
        Snapshot.Match(new {
            TypeObjectSerializer.AllowedTypes,
            TypeObjectSerializer.AllowedTypesByNamespaces,
            TypeObjectSerializer.AllowedTypesByDependencies });
    }

    [Fact]
    public void AddAllowedTypes_AddAllowedTypesByNamespaces_Success()
    {
        // Arrange
        TypeObjectSerializer.Clear();

        // Act
        TypeObjectSerializer.AddAllowedTypes("Mongo", "SwissLife");

        // Assert
        Snapshot.Match(new {
            TypeObjectSerializer.AllowedTypes,
            TypeObjectSerializer.AllowedTypesByNamespaces,
            TypeObjectSerializer.AllowedTypesByDependencies });
    }

    [Fact]
    public void AddAllowedTypes_AddAllowedTypesByTypes_Success()
    {
        // Arrange
        TypeObjectSerializer.Clear();

        // Act
        TypeObjectSerializer.AddAllowedTypes(typeof(Foo), typeof(Bar));

        // Assert
        Snapshot.Match(new {
            TypeObjectSerializer.AllowedTypes,
            TypeObjectSerializer.AllowedTypesByNamespaces,
            TypeObjectSerializer.AllowedTypesByDependencies });
    }

    public class Bar
    {
        public int Id { get; set; }
        public string? BarName { get; set; }
    }

    public class Foo
    {
        public int Id { get; set; }
        public string? FooName { get; set; }
    }
}
