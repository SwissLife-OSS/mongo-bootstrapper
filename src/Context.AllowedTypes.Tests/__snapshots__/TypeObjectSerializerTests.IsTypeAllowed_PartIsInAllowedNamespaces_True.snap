{
  "AllowedTypes": [
    {
      "Key": "MongoDB.Extensions.Context.AllowedTypes.Tests.Foo",
      "Value": true
    }
  ],
  "AllowedTypesByNamespaces": [
    "MongoDB.Extensions.Context"
  ],
  "AllowedTypesByDependencies": []
}
